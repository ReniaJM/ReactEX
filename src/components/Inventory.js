import React from 'react'
import {fbase} from '../firebase'
import BookView from './BookView'



export default class Inventory extends React.Component{

    constructor(){
        super()
        this.state= {
            books: []
        }
    }
    componentDidMount(){
        this.ref= fbase.syncState('bookstore/books',{
            context: this,
            state: 'books'
        });
    }

    componentWillUnmount (){
        fbase.removeBinding( this.ref);
    }


    render(){
        let adminCss = {
            paddingTop: '15px',
        }

        const bookListing = this.state.books.map(book  =>{
             return <BookView key={book.name} book={book} addToOrder = {this.props.addToOrder}/>
        });
        return (
            <div className=" inventory col-md-6" style={adminCss}>
                <h2>Bookstore inventory:</h2>
                {bookListing}
            </div>
        );
    }
}

