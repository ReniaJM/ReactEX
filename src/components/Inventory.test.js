import ReactDOM from "react-dom";
import React from "react";
import Inventory from "./Inventory.js";
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';


configure ({adapter: new Adapter ()});

describe ('Inventory-test', ()=> {
    it('Inventory render without a problem', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Inventory/>, div);
        ReactDOM.unmountComponentAtNode(div);
        // to jets kasowanie diva
    });

    it('hello world renderee', () => {
        const wrapper = shallow(<Inventory/>);
        // console.log(wrapper.debug());
        expect(wrapper.find('div').text('Książki'))
    })

    it('snapshot matches', () => {
        const wrapper = shallow(<Inventory/>);
        // console.log(wrapper.debug());
        expect(wrapper).toMatchSnapshot();
    })

})

