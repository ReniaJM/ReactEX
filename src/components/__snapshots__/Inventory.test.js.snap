// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory-test snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Inventory />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Książki",
      "className": " inventory col-md-4",
      "style": Object {
        "paddingTop": "15px",
      },
    },
    "ref": null,
    "rendered": "Książki",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Książki",
        "className": " inventory col-md-4",
        "style": Object {
          "paddingTop": "15px",
        },
      },
      "ref": null,
      "rendered": "Książki",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
